### ISSUE TITLE
Endless page refreshing "Webseite lädt"

### ISSUE THREAD

**Issue by Ruperr** [neopostmodern/wohnungsbot#203]
Hi,

apparently, the bot is broken again, as it refreshes the page endlessly and doesn't search for flats. It gets stuck on "Webseite lädt" afterwards it says "Geschafft, ich übernehme wieder!" in an infinite loop, until the captcha appears. 

Linux, Wohnungsbot 1.7.1

**Comment by delereme** [comment]
Hi,

I have the same issue. Problem still persists when trying version 1.7.

Windows 10, Wohnungsbot 1.71 an 1.7

**Comment by Quantenoptiker** [comment]
Hi,

same issue here since yesterday. For 3 days before it worked fine.
Wohnungsbot-1.7.1-armv7l on Raspberry PI400

**Comment by CRank1907** [comment]
Same issues since beginning of the week on M1 Mac with Sonoma 14.4.1 and app version 1.7.1

**Comment by neopostmodern** [comment]
Does this issue happen consistently once it appears or does the Bot sometimes work in between? Are there suitable flats in view while the issue happens?

**Comment by CRank1907** [comment]
Yes, it's been there since it appeared, didn't work since then.
Suitable flats are available.

First I used the "non-ARM" version of the app, it worked fine for a couple of days. After the issue came up, I tried the ARM-version. Same issue.

**Comment by Ruperr** [comment]
Surprisingly, the bot seems to work in between. Just started the bot to test it again, now it works flawlessly. Strange!

**Comment by neopostmodern** [comment]
Observed a similar thing, I think. It just said "Geschafft, ich übernehme wieder!" for a long time without doing any applications.

**Comment by CRank1907** [comment]
https://github.com/neopostmodern/wohnungsbot/assets/167512306/94b78ab0-58f2-4f54-955d-202bb402e27a

I tried to upload a video, hope it works and helps :)

**Comment by CRank1907** [comment]
Maybe following screenshots help to identify the problem.
For testing, I chose not to use the immoscout account anymore, but I see that it still uses it and it stays logged in.

Is there any log-file that would help? I'd be happy to help getting it fixed quickly.

<img width="613" alt="Bildschirmfoto 2024-04-23 um 08 23 04" src="https://github.com/neopostmodern/wohnungsbot/assets/167512306/7552a462-45da-4fff-97c9-3bbd3a57b050">

<img width="1103" alt="Bildschirmfoto 2024-04-23 um 08 22 26" src="https://github.com/neopostmodern/wohnungsbot/assets/167512306/180f5d32-786c-47ab-9b63-9b0690be5070">

**Comment by neopostmodern** [comment]
Uh, that seems to be a separate bug! I've opened #204

Unfortunately there are now useful logs at the time (see #116 / #197).

The only chance would be to run it locally and poke around the code.

**Comment by delereme** [comment]
I uploaded a video of the problem. For me this problem has always been happening from the first time using the programme (end of March) with one exeption on 2024-04-23 where it worked and contaced an add. As you can see, there are flats in view; though I can't say if they are suitable to the search criteria because it is reloading to fast.

https://github.com/neopostmodern/wohnungsbot/assets/166100177/3eea4f24-ef49-41f0-8e13-31a367ad8f2a

**Comment by neopostmodern** [comment]
Thanks @delereme for the video! Haven't seen this intensity of refreshes before, crazy stuff. Definitely a bug somewhere that causes the infinite loop.

**Comment by vivol8** [comment]
I'm new to wohnungsbot and github and just installed the app. Sadly came cross this issue where the bot is refreshing the side every second followed bei "geschafft" followed by the "ich bin kein Roboter" until thie pop up is gone and im back on the maine page of immoscout and the "website lädt" / "geschafft" circle. 

Is there any solution to this? Or any tip how i can make it work? 
Have a mac and version 1.8.0

**Comment by neopostmodern** [comment]
@vivol8 does this mean you see the "no bots allowed" / captcha page over and over again?
Could you maybe make a screen recording?

**Comment by vivol8** [comment]
@neopostmodern No, as you can see in the video, it either gets stuck on "not a robot" and then returns to the main page where it freezes between "geschafft" and "website lädt." until its back on the robot page and then all over again. That's all I've got so far. I restarted the program, the bot, and my search, deactivated the "sort by newest" option, deleted the app, and reloaded it. Nothing helped. :/


https://github.com/user-attachments/assets/342e7e5b-725b-418a-96dc-79f1186a9092

**Comment by vivol8** [comment]
@neopostmodern No, as you can see in the video, it either gets stuck on "not a robot" and then returns to the main page where it freezes between "geschafft" and "website lädt." until its back on the robot page and then all over again. That's all I've got so far. I restarted the program, the bot, and my search, deactivated the "sort by newest" option, deleted the app, and reloaded it. Nothing helped. :/

https://github.com/user-attachments/assets/d48449ce-c5be-4ff0-b5a3-cebb36e5c1db

**Comment by tobibrb** [comment]
Hi Guys, I was able to reproduce the bug and think I found what caused it in my case.
Problem seems to appear if there is no `contactDetails` in the `processOverviewDataEntry()` function in `data.ts` (https://github.com/neopostmodern/wohnungsbot/blob/b8db87a1a31874c6e11fe926e6518e1bb17f4581/app/actions/data.ts#L38).

Happened for me while trying to get this: https://www.immobilienscout24.de/expose/154605896.

Can anyone confirm this happens when the contact details are empty?

**Comment by tobibrb** [comment]
Had this working for 1 week now, no further issues so far.

**Comment by neopostmodern** [comment]
@tobibrb thanks for narrowing this down! Unfortunately the link you provided was offline again by the time I checked and I really can't reproduce this. Something feels off about this to me, because they have not one but *two* non-personal salutations in their system (`COMPANY` and `NO_SALUTATION`). So I'm thinking there must be something else going on, that you fixed with your PR, but which feels more like a hack.
Could you maybe try logging which inserts get caught by your change and either share URLs, their JSON data or some specific observation what is missing?

**Comment by Ruperr** [comment]
I haven't used the bot in a long time, but after testing it today, I can confirm that the error still persists in the current version 1.8.0 on Arch Linux.

**Comment by florianmodel** [comment]
Running into the same issue here. From my observation, the bot accelerates the refresh rate over time. It starts as a roughly once every two seconds, but after a few minutes, it's at 4, 5 times a second. 

MacOS 15.3.1

**Comment by solarmaster961** [comment]
Confirming this issue still exists in version `1.8.0`.

My experience matches recent descriptions:
1. The bot logs in successfully.
2. It then gets stuck in a loop, constantly refreshing the page multiple times per second with the message "website lädt".
3. The refresh rate seems to increase over time.
4. Occasionally, a CAPTCHA appears and can be solved as expected, but the bot returns to the refresh loop afterwards.
5. Eventually, I stopped the bot as it didn't seem to progress beyond this loop.

Tested on:
*   Raspberry Pi 3a with 64-bit OS
*   Windows 11

Both systems were running Wohnungsbot v1.8.0.

**Comment by vanchhit** [comment]
Hey! I recently downloaded version 1.8.1 and the Arm (Apple Silicon) version is not working on my M1 Macbook Air (2021). I used the Intel version, which is working, but there is an infinite loop at "Sortierung ändern". Unfortunately I do not currently have other devices to test on, but would love to know if anyone has any fixes, whether temporary or hard-coding-wise.
Thank you so much for the bot though, if and when it starts working, it will be a godsend!

<img width="1198" alt="Image" src="https://github.com/user-attachments/assets/d0ef5003-4294-4ba2-9339-e5d53e1f008a" />

**Comment by FabioSantos14** [comment]
Also having this issue on 1.8.1 running on my Windows 10. If someone has any tips on how to restart it and overcome this issue would be great.

EDIT: Tried to use the restart function on the bot, and tried to close and open the application multiple times without success. After I restarted my computer it started working afterwards, has now been running for 24h without issues.

**Comment by arnaskap** [comment]
I also started getting this behaviour yesterday on version 1.8.1 - first on the OS I usually run the bot on (Fedora), but I also tried running the bot on my Windows 11 and MacOS (Intel CPU) devices, and in both cases it remains stuck refreshing the listings as well.

I'm happy to provide logs or any other info if it would help out.

**Comment by FabioSantos14** [comment]
Just want to update that this only happened once and never happened again for me, it has been running for more than a week without troubles.

**Comment by eefkrr** [comment]
My Wohnungsbot has been stuck in the endless reload loop for two weeks now (M1 Mac). Is there a solution in sight or is there any data I can provide?

**Comment by muschuu** [comment]
July-same issue. Goes into an endless loop with refreshing until its gets to the immoscout are you a human check.

**Comment by stillcrank** [comment]
I could reproduce the bug today by playing with the search filters. Every time I checked the "Unbedingt keine Tauschwohnung" Filter, die reload loop appeared. As soon as I unchecked it, the bot worked fine again. I didn't find any other filter causing the reload loop.

Edit: It is working again by 
- saving the config.json
- uninstalling the bot and deleting the folder in "Roaming"
- installing the bot again
- restoring the config.json with the "noSwapApartment" filter set to true

So maybe the bug is caused by some cache issue?

**Comment by Ciag2** [comment]
I'm still facing this issue.
I tried to follow the steps mentioned, but unfortunately it didn't work for me.
I'm on a Mac. Is anyone else still experiencing this problem?

**Comment by C3VO030** [comment]
I have the same issue on Windows and on the Raspberry Pi.

**Comment by rieck** [comment]
Unfortunately, I can confirm that version 1.8.2 is also affected.

Edit: I can further confirm that this is not a caching issue. I am running a dockerized instance of Wohnungsbot, and the problem persisted even after I removed all data (including all cache files) except for `config.json`.

**Comment by C3VO030** [comment]
It worked for a while, but unfortunately it doesn’t work again at the moment. Has anything changed on the website?

**Comment by Ciag2** [comment]
Is anybody working on this issue? I'am still getting it on macOS. The berlin housing market is hell :)! Thank you.

**Comment by Eugene-kms** [comment]
> Had this working for 1 week now, no further issues so far.
> 
> But keep in mind, I changed quite a lot locally to make your bot work with my city.

Hi!
Could you help me fix this problem "Website lädt", please? 
I was trying many different ways, but it doesn't help me :((
Best regards!

**Comment by tobibrb** [comment]
Hi @Eugene-kms 

If you're comfortable with a bit of dev work, I’d recommend cloning the repo, following the setup guide, and cherry-picking my commit to test it locally.

Just a heads-up: the fix is a little hacky and hasn't been tested (as far as I know). My focus back then was simply getting it to work for my Potsdam setup after a lot of trial and error.

Best of luck!

**Comment by erecluetzkendorf** [comment]
Actually I fixed it a bit easier. The problem itself always started with the specific config.json, not with any other component. After deleting it and creating a new one, it worked again. The only difference is the missing cache. Probably clearing it, might works too. Can't try it right now tho.

**Comment by krono** [comment]
Hi @tobibrb 

> […]
> Just a heads-up: the fix is a little hacky and hasn't been tested (as far as I know). My focus back then was simply getting it to work for my Potsdam setup after a lot of trial and error.
> […]


Did you get it working for Potsdam?
